ui/pages/sample/components/sample-petnames-form.tsx

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import {
+  Box,
+  Text,
+  Button,
+  FormTextField,
+  HelpText,
+  HelpTextSeverity,
+} from '../../../components/component-library';
+import Card from '../../../components/ui/card';
+import {
+  Display,
+  FlexDirection,
+  AlignItems,
+  TextAlign,
+  TextVariant,
+  JustifyContent,
+  BorderRadius,
+  BackgroundColor,
+  TextColor,
+  BlockSize,
+} from '../../../helpers/constants/design-system';
+import { useSamplePetnames } from '../../../ducks/sample-petnames';
+import { useSamplePetnamesForm } from '../hooks/useSamplePetnamesForm';
+
+export function SamplePetnamesForm() {
+  const petnames = useSamplePetnames();
+  const {
+    isSubmitting,
+    submitStatus,
+    isFormValid,
+    formError,
+    handleSubmit,
+    getFieldProps,
+  } = useSamplePetnamesForm();
+
+  const onSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    handleSubmit();
+  };
+
+  return (
+    <Card width={BlockSize.Full} style={{ maxWidth: '500px' }}>
+      <Box
+        display={Display.Flex}
+        flexDirection={FlexDirection.Column}
+        alignItems={AlignItems.center}
+        gap={4}
+      >
+        <Text variant={TextVariant.headingSm}>Pet Names on this network</Text>
+
+        {/* Pet Names List */}
+        <Box
+          display={Display.Flex}
+          flexDirection={FlexDirection.Column}
+          gap={2}
+          width={BlockSize.Full}
+        >
+          {Object.entries(petnames.namesForCurrentChain).length === 0 ? (
+            <Text
+              variant={TextVariant.bodyMd}
+              color={TextColor.textAlternative}
+              textAlign={TextAlign.Center}
+            >
+              No pet names added yet
+            </Text>
+          ) : (
+            Object.entries(petnames.namesForCurrentChain).map(
+              ([addr, name]) => (
+                <Box
+                  key={addr}
+                  display={Display.Flex}
+                  justifyContent={JustifyContent.spaceBetween}
+                  padding={2}
+                  backgroundColor={BackgroundColor.backgroundAlternative}
+                  borderRadius={BorderRadius.LG}
+                >
+                  <Text>{name}</Text>
+                  <Text
```

Comment:
Minor, as we have lots of nesting here and a lot of JSX, should we add some local components to demonstrate it's good practice to keep the components very modular, even in one file.

Such as `PetNameList` and `PetNameListItem`?